numba/tests/test_types.py

```diff
@@ -785,5 +785,60 @@ def f(a):
                              f(Foo()))
 
 
+class TestIssues(TestCase):
+    def test_omitted_type(self):
+        # issue https://github.com/numba/numba/issues/5471
+        def inner(a):
+            pass
+
+        @overload(inner)
+        def inner_overload(a):
+            if not isinstance(a, types.Literal):
+                return
+            return lambda a: a
+
+        @njit
+        def my_func(a='a'):
+            return inner(a)
+
+        @njit
+        def f():
+            return my_func()
+
+        @njit
+        def g():
+            return my_func('b')
+
+        self.assertEqual(f(), 'a')
+        self.assertEqual(g(), 'b')
+
+    def test_type_of_literal(self):
+        # type(val) where val is a literal should not give a literal type.
+        def inner(a):
+            pass
+
+        @overload(inner)
+        def inner_overload(a):
+            if not isinstance(a, types.Literal):
+                return
+            self.assertIsInstance(a, types.Literal)
+            return lambda a: type(a)(a + 1)
```

Comment:
As this is a type cast, would it be better to do `(a + 1.23)` or similar, to demonstrate the cast has an effect, else the changes to `type` impl could just be a `nop`?